Tests/MapboxNavigationTests/LaneViewTests.swift

```diff
@@ -0,0 +1,194 @@
+        
+        XCTAssertEqual(([.straightAhead, .slightLeft] as LaneIndication).ranked(favoring: nil),
+                       .init(primary: .straightAhead, secondary: .left))
+        XCTAssertEqual(([.straightAhead, .slightLeft] as LaneIndication).ranked(favoring: .right),
+                       .init(primary: .straightAhead, secondary: .left))
+        XCTAssertEqual(([.straightAhead, .slightLeft] as LaneIndication).ranked(favoring: .left),
+                       .init(primary: .straightAhead, secondary: .left))
+        XCTAssertEqual(([.straightAhead, .slightLeft] as LaneIndication).ranked(favoring: .sharpLeft),
+                       .init(primary: .straightAhead, secondary: .left))
+        XCTAssertEqual(([.straightAhead, .slightLeft] as LaneIndication).ranked(favoring: .slightLeft),
+                       .init(primary: .left, secondary: .straightAhead))
+        
+        XCTAssertEqual(([.straightAhead, .right] as LaneIndication).ranked(favoring: nil),
+                       .init(primary: .straightAhead, secondary: .right))
+        XCTAssertEqual(([.straightAhead, .right] as LaneIndication).ranked(favoring: .left),
+                       .init(primary: .straightAhead, secondary: .right))
+        XCTAssertEqual(([.straightAhead, .right] as LaneIndication).ranked(favoring: .right),
+                       .init(primary: .right, secondary: .straightAhead))
+        XCTAssertEqual(([.straightAhead, .right] as LaneIndication).ranked(favoring: .sharpRight),
+                       .init(primary: .right, secondary: .straightAhead))
+        XCTAssertEqual(([.straightAhead, .right] as LaneIndication).ranked(favoring: .slightRight),
+                       .init(primary: .right, secondary: .straightAhead))
+        
+        XCTAssertEqual(([.straightAhead, .slightRight] as LaneIndication).ranked(favoring: nil),
+                       .init(primary: .straightAhead, secondary: .right))
+        XCTAssertEqual(([.straightAhead, .slightRight] as LaneIndication).ranked(favoring: .left),
+                       .init(primary: .straightAhead, secondary: .right))
+        XCTAssertEqual(([.straightAhead, .slightRight] as LaneIndication).ranked(favoring: .right),
+                       .init(primary: .straightAhead, secondary: .right))
+        XCTAssertEqual(([.straightAhead, .slightRight] as LaneIndication).ranked(favoring: .sharpRight),
+                       .init(primary: .straightAhead, secondary: .right))
+        XCTAssertEqual(([.straightAhead, .slightRight] as LaneIndication).ranked(favoring: .slightRight),
+                       .init(primary: .right, secondary: .straightAhead))
+        
+        XCTAssertEqual(([.left, .uTurn] as LaneIndication).ranked(favoring: nil),
+                       .init(primary: .left, secondary: nil))
+        XCTAssertEqual(([.left, .uTurn] as LaneIndication).ranked(favoring: .left),
+                       .init(primary: .left, secondary: nil))
```

Comment:
Read this line as: if the lane has left and U-turn indications, and the maneuver calls for turning left, draw a left turn. (Ideally, we’d draw a U-turn-or-left icon, but we don’t have an asset for that, so we instead prefer the lane use matching the maneuver.)